flink-python/src/main/java/org/apache/flink/table/runtime/operators/python/AbstractPythonScalarFunctionOperator.java

```diff
@@ -145,13 +148,21 @@ public void processElement(StreamRecord<IN> element) throws Exception {
 	}
 
 	@Override
-	public PythonFunctionRunner<IN> createPythonFunctionRunner() {
+	public PythonEnv getPythonEnv() {
+		return scalarFunctions[0].getPythonFunction().getPythonEnv();
+	}
+
+	@Override
+	public PythonFunctionRunner<IN> createPythonFunctionRunner() throws IOException {
```

Comment:
The IOException could be removed if createBaseDirectory is done in the open method of ProcessEnvironmentManager.

Reply:
The PythonDependencyInfo#create still throws IOException so we can not remove it currently.